src/App.tsx

```diff
@@ -1,24 +1,146 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState, useEffect, useMemo } from 'react';
 import './App.css';
+import Navigation from "./components/Navigation";
+import Scores from "./components/Scores";
+import Home from './components/Home';
+import Total from "./components/Total";
+import {BrowserRouter as Router, Switch, Route} from "react-router-dom";
+import {IPlayer, IScore, IRoutes} from "./types";
+import ROUTES from './routes';
+import {nanoid} from 'nanoid';
+import {debounce} from 'debounce';
+
+const scoreTemplate = {
+	military: 0,
+	treasury: 0,
+	wonders: 0,
+	civilian: 0,
+	commerce: 0,
+	guild: 0,
+	compass: 0,
+	tablet: 0,
+	gear: 0
+};
 
 function App() {
+	const [players, setPlayers] = useState<IPlayer[]>([]);
+
+	useEffect(() => {
+		saveToStorage(players);
+	});
+
+	useEffect(() => {
+		restoreFromStorage();
+	}, []);
+
+	const saveToStorage = useMemo(() => debounce((players: IPlayer[]) => {
+		localStorage.setItem('players', JSON.stringify(players));
```

Comment:
It might be easier to save players only when 'players' state has been changed, not on every render with debouncing.

Reply:
Fixed